src/main/javascript/src/pages/interest/create/create.js

```diff
@@ -25,6 +25,10 @@ export class setInterestFormFields {
 			department: departmentChoice,
 			keyword: interestKeyword
 		}
-		this.interestManager.addInterest(this.myInterest).then(()=>{this.router.navigateToRoute("view interests")});
+		this.interestManager.addInterest(this.myInterest).then(()=>{
+			this.departmentManager.getDepartments().then(response => {
+				this.departments = response;
+			});
+		});
```

Comment:
adding interests doesn't update the departments, does it? If it does, a better option might be to return the updated list of departments with the added interest in the response of adding an interest. This would save some time having to make another call to the back end.

Perhaps don't worry about it now. We're out of time.